Tests/SentryTests/SentryCrash/SentryCrashJSONCodec_Tests.m

```diff
@@ -463,6 +463,55 @@ - (void)testSerializeDeserializeDictionaryMultipleEntries
     XCTAssertEqualObjects(result, original, @"");
 }
 
+- (void)testSerializeDeserializeDictionaryIntKeys
+{
+    NSString *expectedJson = @"{\"1\":\"Value\",\"2\":1000,\"3\":true}";
+    NSDictionary *expected = @{ @"1" : @"Value", @"2" : @1000, @"3" : @YES };
+    NSDictionary *original = @{ @1 : @"Value", @3 : @YES, @2 : @1000 };
+
+    [self testSerializeDeserializeDictionaryWith:original
+                                     andExpected:expected
+                                 andExpectedJSON:expectedJson];
+}
+
+- (void)testSerializeDeserializeDictionaryWith:(NSDictionary *)original
+                                   andExpected:(NSDictionary *)expected
+                               andExpectedJSON:(NSString *)expectedJson
+{
+    NSError *error = (NSError *)self;
+    NSString *jsonString = toString([SentryCrashJSONCodec encode:original
+                                                         options:SentryCrashJSONEncodeOptionSorted
+                                                           error:&error]);
+    XCTAssertNotNil(jsonString);
+    XCTAssertNil(error);
+    XCTAssertEqualObjects(jsonString, expectedJson);
+
+    NSDictionary *result = [SentryCrashJSONCodec decode:toData(jsonString) options:0 error:&error];
+    XCTAssertNotNil(result);
+    XCTAssertNil(error);
+    XCTAssertEqualObjects(result, expected);
```

Comment:
M: why don't we split encode/decode in 2 different tests?

Reply:
I agree with you fully here. I don't like how the tests are written here. Still, I want to follow the test pattern in this test class here. Keep in mind that this is SentryCrash and anyway at some point, we want to get rid of it because it is outdated.